app/services/LocationService.js

```diff
@@ -383,70 +423,93 @@ export default class LocationServices {
+                    if (isPlatformAndroid()) {
+                      // showLocationSettings() only works for android
+                      BackgroundGeolocation.showLocationSettings();
+                    } else {
+                      // eslint-disable-next-line
+                      Linking.openURL('App-Prefs:Privacy'); // Deeplinking method for iOS
+                    }
+                  },
+                },
+                {
+                  text: languages.t('label.no'),
+                  onPress: () => {
+                    console.log('No Pressed'), LocationServices.stop();
+                  },
+                  style: 'cancel',
+                },
+              ],
+            ),
+          1000,
+        );
       } else if (!status.authorization) {
         // we need to set delay or otherwise alert may not be shown
-        // setTimeout(
-        //   () =>
-        //     Alert.alert(
-        //       languages.t('label.require_location_information_title'),
-        //       languages.t('label.require_location_information_message'),
-        //       [
-        //         {
-        //           text: languages.t('label.yes'),
-        //           onPress: () => BackgroundGeolocation.showAppSettings(),
-        //         },
-        //         {
-        //           text: languages.t('label.no'),
-        //           onPress: () => console.log('No Pressed'),
-        //           style: 'cancel',
-        //         },
-        //       ],
-        //     ),
-        //   1000,
-        // );
+        setTimeout(
+          () =>
+            Alert.alert(
+              languages.t('label.require_location_information_title'),
+              languages.t('label.require_location_information_message'),
+              [
+                {
+                  text: languages.t('label.yes'),
+                  onPress: () => BackgroundGeolocation.showAppSettings(),
+                },
+                {
+                  text: languages.t('label.no'),
+                  onPress: () => {
+                    console.log('No Pressed'), LocationServices.stop();
+                    //                     Settings.setIsLogging(false);
+                  },
+                  style: 'cancel',
+                },
+              ],
+            ),
+          1000,
+        );
       }
     });
   }
 
+  static isheadlessAlreadyRunning() {
+    return BackgroundGeolocation.getConfig(config => {
+      if (!config.stopOnTerminate && config.notificationsEnabled) return true;
+    });
+  }
+
   static stop() {
     // unregister all event listeners
-    PushNotification.localNotification({
-      title: languages.t('label.location_disabled_title'),
-      message: languages.t('label.location_disabled_message'),
-    });
     BackgroundGeolocation.removeAllListeners();
     BackgroundGeolocation.stop();
 
+    BackgroundGeolocation.checkStatus(({ locationServicesEnabled }) => {
+      if (!locationServicesEnabled) {
```

Comment:
Since the only thing different between these two statements is the message, i think it would read a bit cleaner like this 

```js
const message = locationServicesEnabled
  ? languages.t('label.location_disabled_message')
  : languages.t('label.location_stopped_message');
        
PushNotification.localNotification({
  id: LOCATION_DISABLED_NOTIFICATION,
  title: languages.t('label.location_disabled_title'),
  message,
});
```